tests/flows/test_vesting.py

```diff
@@ -106,22 +139,127 @@ def test_main(
 
     # Tests permissions
     perms = erc20_token.get_permissions(consumer_wallet.address)
-    assert perms[0] is True
+    assert perms[RolesERC20.MINTER] is True
+    assert perms[RolesERC20.PAYMENT_MANAGER] is False
+
+    erc20_token.add_payment_manager(consumer_wallet.address, consumer_wallet)
+    perms = erc20_token.get_permissions(consumer_wallet.address)
+    assert perms[RolesERC20.PAYMENT_MANAGER] is True
+
+
+@pytest.mark.unit
+def test_minting_failed_after_deploying_pool(
+    web3, config, publisher_wallet, consumer_wallet, factory_router
+):
+    """Tests failure to mint new erc20 token even if consumer has the minter role after deploying a pool."""
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3,
+        config,
+        erc721_publisher=publisher_wallet,
+        erc20_minter=consumer_wallet,
+        cap=to_wei(1000),
+    )
+    erc721_nft.add_to_create_erc20_list(consumer_wallet.address, publisher_wallet)
 
     # Tests consumer deploys pool and check market fee
-    initial_ocean_liq = to_wei("0.02")
+    mint_fake_OCEAN(config)
+    initial_ocean_liq = to_wei(200)
     ocean_contract = ERC20Token(web3=web3, address=get_address_of_type(config, "Ocean"))
-    ocean_contract.approve(
-        get_address_of_type(config, "Router"), to_wei("0.02"), consumer_wallet
+    ocean_contract.approve(factory_router.address, initial_ocean_liq, consumer_wallet)
+
+    erc20_token.deploy_pool(
```

Comment:
There are 7 calls to `erc20_token.deploy_pool()`, each mostly with the same parameters. Each call takes 17 lines, for 119 lines total. This is a DRY violation. Better: create a new function at the bottom of test_vesting.py called `_deploy_pool()` that takes in a couple parameters, hardcodes the rest, and calls `erc20_token.deploy_pool()` there. Then, then replace all the calls to `erc20_token.deploy_pool()` with the new `_deploy_pool()` method.

Reply:
And, there are 10 calls to `erc721_factory.deploy_erc721_contract()`. DRY violation. Fix it similarly, please.